### PR TITLE
chore(expo): update expo version to 50

### PR DESCRIPTION
**Pull request by d4v3d4vE**
- removed `sdkVersion` and `privacy` section from `app.json` because it is necessary when version checking with `npx expo-doctor`
- updated required packages
- https://expo.dev/changelog/2024/01-18-sdk-50

- had an Issue with this update, fix: https://github.com/expo/expo/issues/28554#issuecomment-2162969959

SVA-1305

For this update you'll need to Update your Node Version to v18.19.1. 
To update Node.js on your PC, follow these steps based on your operating system:

#### For Windows:

1. **Using Node Version Manager (nvm) for Windows**:
   - Download and install nvm for Windows from the [nvm-windows GitHub repository](https://github.com/coreybutler/nvm-windows/releases).
   - Open Command Prompt and run the following commands:
     ```sh
     nvm install latest
     nvm use latest
     ```
   - This will install the latest version of Node.js and set it as the active version.

2. **Manual Update**:
   - Go to the [Node.js download page](https://nodejs.org/).
   - Download the Windows installer for the latest version.
   - Run the installer and follow the on-screen instructions to complete the installation. The installer will automatically replace the old version.

#### For macOS:

1. **Using Homebrew**:
   - Open Terminal and run the following commands:
     ```sh
     brew update
     brew upgrade node
     ```

2. **Using Node Version Manager (nvm)**:
   - If you don't have `nvm` installed, install it by running:
     ```sh
     curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.2/install.sh | bash
     source ~/.nvm/nvm.sh
     ```
   - Then, to install and use the latest version of Node.js:
     ```sh
     nvm install node
     nvm use node
     ```

3. **Manual Update**:
   - Go to the [Node.js download page](https://nodejs.org/).
   - Download the macOS installer for the latest version.
   - Run the installer and follow the on-screen instructions to complete the installation. The installer will automatically replace the old version.

#### For Linux:

1. **Using Node Version Manager (nvm)**:
   - Install `nvm` if you don't have it:
     ```sh
     curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.2/install.sh | bash
     source ~/.nvm/nvm.sh
     ```
   - Then, to install and use the latest version of Node.js:
     ```sh
     nvm install node
     nvm use node
     ```

2. **Using Package Manager**:
   - For Debian-based systems (Ubuntu):
     ```sh
     sudo apt update
     sudo apt install -y nodejs
     ```
   - For Red Hat-based systems (Fedora):
     ```sh
     sudo dnf module install -y nodejs:latest
     ```

By following these steps, you should be able to update Node.js to the latest version on your PC.

### CODE REVIEW

File: app.json [hunk 1]
```diff
@@ -3,7 +3,6 @@
     "name": "Smart Village App",
     "slug": "smart-village-app",
     "owner": "ikusei",
-    "privacy": "unlisted",
     "description": "An open source React Native app for smart villages merging regional offers and services in one place - in your pocket.",
```
Comment (ardasnturk) [inline app.json:6]: Have you tried using other values instead of deleting privacy? Like `hidden`, for example.

File: eas.json [hunk 2]
```diff
@@ -4,7 +4,7 @@
   },
   "build": {
     "production": {
-      "node": "16.13.0",
+      "node": "v18.19.1",
       "yarn": "1.22.19",
```
Comment (ardasnturk) [inline eas.json:8]: Should we upgrade the node version on our computer for this update?

**Review by ardasnturk — COMMENTED** [review]
Have you read the changelogs of all updated packages? No breaking change affecting our app?

File: eas.json (hunk 2 above)
Comment (d4v3d4vE) [reply eas.json:8]: yes

**Comment by d4v3d4vE** [comment]
No breaking change affecting our app.

File: eas.json (hunk 2 above)
Comment (ardasnturk) [reply eas.json:8]: Can you please briefly explain how to do this in PR?

File: app.json (hunk 1 above)
Comment (d4v3d4vE) [reply app.json]: no I have not.

File: eas.json (hunk 2 above)
Comment (d4v3d4vE) [reply eas.json:8]: done

File: app.json (hunk 1 above)
Comment (d4v3d4vE) [reply app.json]: but it's already added back, therefore not an issue anymore

**Review by ardasnturk — APPROVED** [review]

File: eas.json (hunk 2 above)
Comment (donni106) [reply eas.json:8]: or with [asdf](https://asdf-vm.com/): `asdf install nodejs 18.19.1`

**Review by donni106 — APPROVED** [review]

File: eas.json [hunk 3]
```diff
@@ -4,7 +4,7 @@
   },
   "build": {
     "production": {
-      "node": "16.13.0",
+      "node": "v18.19.1",
```
Comment (donni106) [inline eas.json:7]: is it fine to have the version `v` letter added here now?

File: metro.config.js [hunk 4]
```diff
@@ -1,22 +1,14 @@
...
-          inlineRequires: true
-        }
-      })
-    },
-    resolver: {
-      assetExts: [...assetExts, 'mtl', 'obj', 'vrx']
-    }
-  };
-})();
+config.resolver.sourceExts.push('mtl', 'obj', 'vrx');
```
Comment (donni106) [inline metro.config.js:6]: what was `assetExts` is now `sourceExts`?

File: eas.json (hunk 3 above)
Comment (ardasnturk) [reply eas.json:7]: I missed this. There's not supposed to be a `v`!

https://docs.expo.dev/build/eas-json/#selecting-build-tool-versions

image

File: metro.config.js (hunk 4 above)
Comment (ardasnturk) [reply metro.config.js:6]: You're right. It was supposed to be `assetExts` 👍

Comment (ardasnturk) [reply metro.config.js]: https://github.com/smart-village-solutions/smart-village-app-app/pull/1136/commits/dcd0f95e12d1e2f4557f848fddb0c382d036eda5

File: eas.json (hunk 3 above)
Comment (ardasnturk) [reply eas.json]: https://github.com/smart-village-solutions/smart-village-app-app/pull/1136/commits/c577e0b584718f35904aa20681b54f8c7bbc35ae

**Pull request merged**